packages/loader/container-utils/src/error.ts

```diff
@@ -107,7 +106,7 @@ export class DataProcessingError extends LoggingError implements IErrorBase {
             : wrapError(originalError, newErrorFn);
 
         if (message !== undefined) {
-            annotateError(error, extractSafePropertiesFromMessage(message));
+            annotateErrorObject(error, { props: extractSafePropertiesFromMessage(message) });
```

Comment:
What happens if originalError has error code, but is frozen? Should code like this not care about it?
Currently, if I got it right, it would assert.

Reply:
Yeah I debated about this and chose to not optimize for the Frozen case. I just want the code to be robust and explicit.

Suppose we start with an `isFluidError` check and return right away if so... but only if no props were supplied... unless the object implemented addTelemetryProperties.

So didn't seem worth it to try to support that, since the Frozen checks are purely defensive/hypothetical anyway.

Reply:
Update: not supporting frozen objects in any API here. We'll see in telemetry if this ever happens - I highly doubt it.